src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/device-list/device-list.component.html

```diff
@@ -13,6 +13,8 @@
 
 <ng-template #lifeExpectancy
              let-value="value">
+  <span *ngIf="!value.life_expectancy_enabled"
+        i18n>N/A</span>
```

Comment:
Rather than dealing with `null` to `N/A` converstions case by case, what about implementing this in the datatable behaviour itself? I remember that @avanthakkar did this in other places, so I'd rather avoid doing this ad-hoc and implement it as a default behaviour of the column/cell template. What do you think?

Reply:
@epuertat do you mean to put a new property, e.g. `notAvailable`, in `TableColumn` class ?

Reply:
@torchiaf: the component in which the Dashboard datatable is based on, [ngx-datatable](https://swimlane.gitbook.io/ngx-datatable/api/column/inputs#pipe-pipetransform), supports a `pipe` property for each column. We already have a `notAvailable` pipe that converts an empty string to `n/a`:

https://github.com/ceph/ceph/blob/f4df4090b040f1d2c2912533f0cbf5fae6fab9c4/src/pybind/mgr/dashboard/frontend/src/app/shared/pipes/not-available.pipe.ts#L5-L15

Instead of implementing this same behaviour case by case, IMHO we should probably:
- Add that pipe by default to the cd-datatable `pipe` property,
- And extend that pipe to also check for `undefined` or `null` and display the `n/a` on those cases.

Otherwise, we'll end up having to patch every single place where an empty/`null`/`undefined` string may appear.

Reply:
@epuertat I agree with you to put a default value when the value of a cell is `null` or `undefined`.
In our case however, I think the It is different from the general case.
The value of `lifeExpectancy` column depends on many values (`min`, `max`, `life_expectancy_enabled`), it is not a simple translation `null` -> `N/A`.
My fix is supposed to works this way  :
  - when `min` and `max` are empty, show empty value, because it means there are no enough data to calculate the `lifeExpectancy` value
- when `life_expectancy_enabled` is false show `N/A`, because it means that the life expectancy module is disable.

Probably, I should rename `life_expectancy_enabled` -> `life_expectancy_module_enabled`.

What do you think about it?

Reply:
IMO the column we're talk about is too specific to use generic pipes. In that case the implementation of a template within the component is right.

What can be done to limit the number of variations (`N/A` vs `n/a`) which are sadly spread across the whole code is using this:

```
<span *ngIf="!value.life_expectancy_enabled" i18n>{{ "" | notAvailable }}</span> 
```

Reply:
@votdev suggestion is good enough for this!